File: Competitive/Python/enterview.py
#!/usr/bin/env python3
'''
  @see: HackerRank -> Data Structures -> Advanced -> Direct Connections
  @author: Josh Snider
  TODO: Currently times out. time says it takes 2.44 seconds with input5.
'''


def main():
  numtests = int(input(''))
  for _ in range(numtests):
    numcities = int(input(''))
    coords = [int(s) for s in input('').split()]
    pops = [int(s) for s in input('').split()]
    cities = zip(coords,pops)
    cities = sorted(cities)
    cost = 0
    count = len(cities)
    weights = {}
    for x in range(count - 1):
      for y in range(x+1, count):
        #(d1, p1) = city1
        #(d2, p2) = city2
        dist = abs(cities[x][0] - cities[y][0])
        size = max(cities[x][1], cities[y][1])
        cost += dist * size
    print(cost % 1000000007)

if __name__ == '__main__':
  main()

File: Competitive/CPP/EnterView.cpp
/*
  see: HackerRank -> Data Structures -> Advanced -> Direct Connections
  Author: Josh Snider
  This is faster than the python, but both too slow and wrong.
*/
#include <iostream>
#include <sstream>
#include <string>
#include <vector>

std::vector<int>
getIntLine() {
  std::string line;
  std::getline(std::cin, line);
  std::istringstream in(line, std::istringstream::in);
  int n;
  std::vector<int> v;
  while (in >> n) {
    v.push_back(n);
  }
  return v;
}

int main() {
  int numtests = getIntLine()[0];
  for (int x = 0; x < numtests; x++) {
    int numcities = getIntLine()[0];
    auto coords = getIntLine();
    auto pops = getIntLine();
    int cost = 0;
    for (int f = 0; f < numcities; f++) {
      for (int s = f + 1; s < numcities; s++) {
        int dist = std::abs(coords[f]-coords[s]);
        int pipes = std::max(pops[f], pops[s]);
        cost += dist * pipes;
        if (cost >=  1000000007) {
            cost -=  1000000007;
        }
      }
    }
    std::cout << cost << std::endl;
  }
  return 0;
}
